pandas/io/sql.py

```diff
@@ -23,6 +23,7 @@
 import warnings
 
 import numpy as np
+from sqlalchemy.sql.expression import text
```

Comment:
I think this will break without sqlalchemy installed, see pandasSQL_builder where import_optional_dependency is used.

Maybe a better option is to move the conversion to `text` into SQLDatabase.execute, since that will always use SQLAlchemy.

Alternatively, it looks like _convert_params is never called with sqlalchemy=True, so maybe just remove that code path?